python/cuda_cccl/cuda/cccl/parallel/experimental/_bindings_impl.pyx

```diff
@@ -353,43 +383,109 @@ cdef class Enumeration_CCCLType(IntEnumerationBase):
+    MODULUS = Enumeration_OpKind.make_member("MODULUS", cccl_op_kind_t.CCCL_MODULUS)
+    EQUAL_TO = Enumeration_OpKind.make_member("EQUAL_TO", cccl_op_kind_t.CCCL_EQUAL_TO)
+    NOT_EQUAL_TO = Enumeration_OpKind.make_member("NOT_EQUAL_TO", cccl_op_kind_t.CCCL_NOT_EQUAL_TO)
+    GREATER = Enumeration_OpKind.make_member("GREATER", cccl_op_kind_t.CCCL_GREATER)
+    LESS = Enumeration_OpKind.make_member("LESS", cccl_op_kind_t.CCCL_LESS)
+    GREATER_EQUAL = Enumeration_OpKind.make_member("GREATER_EQUAL", cccl_op_kind_t.CCCL_GREATER_EQUAL)
+    LESS_EQUAL = Enumeration_OpKind.make_member("LESS_EQUAL", cccl_op_kind_t.CCCL_LESS_EQUAL)
+    LOGICAL_AND = Enumeration_OpKind.make_member("LOGICAL_AND", cccl_op_kind_t.CCCL_LOGICAL_AND)
+    LOGICAL_OR = Enumeration_OpKind.make_member("LOGICAL_OR", cccl_op_kind_t.CCCL_LOGICAL_OR)
+    LOGICAL_NOT = Enumeration_OpKind.make_member("LOGICAL_NOT", cccl_op_kind_t.CCCL_LOGICAL_NOT)
+    BIT_AND = Enumeration_OpKind.make_member("BIT_AND", cccl_op_kind_t.CCCL_BIT_AND)
+    BIT_OR = Enumeration_OpKind.make_member("BIT_OR", cccl_op_kind_t.CCCL_BIT_OR)
+    BIT_XOR = Enumeration_OpKind.make_member("BIT_XOR", cccl_op_kind_t.CCCL_BIT_XOR)
+    BIT_NOT = Enumeration_OpKind.make_member("BIT_NOT", cccl_op_kind_t.CCCL_BIT_NOT)
+    IDENTITY = Enumeration_OpKind.make_member("IDENTITY", cccl_op_kind_t.CCCL_IDENTITY)
+    NEGATE = Enumeration_OpKind.make_member("NEGATE", cccl_op_kind_t.CCCL_NEGATE)
+    MINIMUM = Enumeration_OpKind.make_member("MINIMUM", cccl_op_kind_t.CCCL_MINIMUM)
+    MAXIMUM = Enumeration_OpKind.make_member("MAXIMUM", cccl_op_kind_t.CCCL_MAXIMUM)
+
+    @staticmethod
+    cdef IntEnumerationMember make_member(str member_name, cccl_op_kind_t op_kind):
         return IntEnumerationMember(
-            type(self),
-            self.enum_name,
-            prop_name,
-            cccl_op_kind_t.CCCL_STATEFUL
+            Enumeration_OpKind,
+            "OpKindEnum",
+            member_name,
+            op_kind
         )
 
 
-    @property
-    def STATELESS(self):
-        return self._stateless
+class OpKind(Enum):
+    """Enumeration of well-known operations for CCCL parallel algorithms.
 
-    @property
-    def STATEFUL(self):
-        return self._stateful
+    This enum provides type-safe access to well-known operations that can be used
+    with parallel algorithms like reduce, scan, etc. These can provide better
+    performance than user-defined callable equivalents by a avoiding the need for
+    compilation, and enabling specialized implementations.
+
+    Example:
+        >>> import cuda.cccl.parallel.experimental as parallel
+        >>>
+        >>> # Use well-known operations with algorithms
+        >>> parallel.reduce_into(data, output, OpKind.PLUS, num_items, init)
+        >>> parallel.reduce_into(data, output, OpKind.MINIMUM, num_items, init)
+    """
+
+    # Arithmetic operations
+    PLUS = Enumeration_OpKind.PLUS
```

Comment:
```suggestion
    PLUS = Enumeration_OpKind.PLUS.value
```

Since `Enumeration_OpKind.PLUS` has type `IntEnumerationMember`, the `OpKind.PLUS` double-packs the value:

```
>>> import cuda.cccl.parallel.experimental._bindings as b
>>> b.OpKind.PLUS
<OpKind.PLUS: <OpKindEnum.PLUS: 2>>
>>> _.value
<OpKindEnum.PLUS: 2>
>>> _.value
2
>>> 
```

With this change the `OpKind.PLUS` would just become `<OpKind.PLUS: 2>`. We would also need to change `class OpKind(Enum):` to `class OpKind(enum.IntEnum):`

Reply:
I implemented this in https://github.com/NVIDIA/cccl/pull/5578/commits/9a359a1637aae3e8ca596715f665a3f0fd9e3da7. Unfortunately, it effectively requires a redefinition of the enum in `bindings.pyi` to work. Using `= int` doesn't give the desired behaviour, where it indicates that the member is an object of type `type[int]`.

Reply:
OK, never mind. The fix is apparently to use a `_value_` type annotation specifying the type of the members: https://github.com/NVIDIA/cccl/pull/5578/commits/263791c29b81af8402ea26f8203371ddfdadedb8

See end of this section: https://typing.python.org/en/latest/spec/enums.html#member-values